docs/functionaldatatypes/validation.md

```diff
@@ -0,0 +1,286 @@
+---
+id: validation
+title: "Validation"
+---
+
+A `Validation[E, A]` is a data type that is either a `Success` with a value of type `A` or a `Failure` with one or more errors of type `E`.
+
+You can think of a `Validation` as looking something like this:
+
+```scala mdoc
+import zio.NonEmptyChunk
+
+sealed trait Validation[+E, +A]
+
+object Validation {
+  case class Success[+A](value: A) extends Validation[Nothing, A]
+  case class Failure[+E](errors: NonEmptyChunk[E]) extends Validation[E, Nothing]
+}
+```
+
+Thus, a `Validation` looks very much like an `Either` except that it can contain multiple errors instead of just one.
+
+The distinguishing feature of `Validation` is that it can accumulate multiple errors. This makes it extremely convenient for problems like data validation where we want to accumulate all validation errors and report them back instead of failing immediately on the first validation error.
+
+To see this, let's consider a simple example of validating a `Person` data type.
+
+```scala mdoc
+case class Person(name: String, age: Int)
+```
+
+The business requirements are that a valid person must have a name that is not empty and an age that is greater than zero.
+
+We could try to model validating a `Person` using `Either` like this:
+
+```scala mdoc
+def validateName(name: String): Either[String, String] =
+  if (name.isEmpty) Left("Name was empty")
+  else Right(name)
+
+def validateAge(age: Int): Either[String, Int] =
+  if (age <= 0) Left(s"Age $age was less than zero")
+  else Right(age)
+
+def validatePerson(name: String, age: Int): Either[String, Person] =
+  for {
+    name <- validateName(name)
+    age  <- validateAge(age)
+  } yield Person(name, age)
+```
+
+However, there is a problem with this. If we try to validate a `Person` with both an invalid `name` and an invalid `age` we will just get a single failure indicating that the name was empty.
+
+This is because `Either` embodies "fail fast" semantics where if one part of a computation fails the subsequent parts of the computation will never be performed.
+
+Normally this is what we want because there is no point in doing further work if a computation has already failed. This is similar to how in traditional imperative programs if an exception is thrown on one line then further lines will not be executed.
+
+However, it is not what we want here.
+
+We could imagine that the `name` and `age` were transmitted by an internal business customer. They would probably not be happy if we told them the name was invalid, they corrected that and sent it again, and then we failed again because the age was invalid.
```

Comment:
pedantic: you may replace "internal customer" and "internal business customer" with "customer" and "business customer" as I don't see the benefit of classifying the customer as such.

Reply:
Done.